lighthouse-viewer/app/src/main.js

```diff
@@ -48,7 +48,11 @@ function main() {
   window.viewer = new LighthouseReportViewer();
 }
 
-main();
+// Defer main until the entire script is loaded.
+// This is because the current build system for the viewer concats files in alphabetical order,
+// which means "psi-api.js" comes _after_ "main.js", causing the constructor to error if `main`
+// is called synchronously.
+Promise.resolve().then(main);
```

Comment:
🤦‍♂

will be following up with commonjs'ing the viewer.

Reply:
> will be following up with commonjs'ing the viewer.

I think this will be significantly less fun than it looks. And viewer is so simple (and we change it so rarely) I don't think it's worth it. Just make `main` happen on `DOMContentLoaded` like any old fashioned web page?

Reply:
> I think this will be significantly less fun than it looks.

I'm not reaching the same conclusion. Seems simple.

Reply:
talked offline. commonjs is a node  thing. viewer is for the web and should not use node conventions.

real solution is to ensure none of the files _do_ anything, and kick off the main loading stuff in index.html